docs/runbooks/VirtualMachineStuckInUnhealthyState.md

```diff
@@ -0,0 +1,307 @@
+2. **Create missing storage classes** or configure default storage
+3. **Resolve PVC/DataVolume failures**:
+   ```bash
+   kubectl get pvc -n <namespace>
+   kubectl describe pvc <pvc-name> -n <namespace>
+   ```
+
+### Image Issues  
+1. **Verify image accessibility**:
+   ```bash
+   # Validate from the node
+   kubectl debug node/<node-name> -it --image=busybox
+
+   # Inside the debug pod, detect runtime and pull
+   ps aux | grep -E "(containerd|dockerd|crio)"
+
+   # For CRI-O/containerd clusters:
+   crictl pull <image-name>
+
+   # For Docker-based clusters (less common):
+   docker pull <image-name>
+
+   exit
+   ```
+2. **Configure image pull secrets** if needed:
+   ```bash
+   kubectl create secret docker-registry <secret-name> \
+     --docker-server=<registry-url> \
+     --docker-username=<username> \
+     --docker-password=<password>
+   ```
+
+### Scheduling Issues
+1. **Review VM scheduling constraints** and relax if too restrictive:
+   - nodeSelector, affinity, and tolerations
+   - Required CPU model, host devices, or features
+
+2. **Verify node taints and tolerations** allow scheduling:
+   - Ensure the VM tolerates node taints that apply to target nodes
+
+3. **Ensure nodes have required capabilities**:
+   - KVM availability, CPU features, GPU, SR-IOV, or storage access
+
+4. If nodes were intentionally cordoned for maintenance, **uncordon** when
+   appropriate:
+   ```bash
+   kubectl uncordon <node-name>
+   ```
+
+### Configuration Issues Resolution
+1. **Fix VM specification errors** based on kubectl describe output:
+   ```bash
+   # Edit VM specification directly
+   kubectl edit vm <vm-name> -n <namespace>
+   
+   # Or patch specific fields
+   kubectl patch vm <vm-name> -n <namespace> --type='merge' \
+     -p='{"spec":{"template":{"spec":{"domain":{"resources": \
+       {"requests":{"memory":"2Gi"}}}}}}}'
+   ```
+2. **Create missing secrets/configmaps**:
+   ```bash
+   kubectl create secret generic <secret-name> \
+     --from-literal=key=value
+   ```
+3. **Adjust resource requests** if they exceed node capacity
+
+### Emergency Workarounds
+- **Restart the VM** to apply specification changes:
+  ```bash
+  # Restart the VM to pick up spec changes
+  virtctl restart <vm-name> -n <namespace>
+  ```
+- **Scale down non-critical workloads** temporarily if resource
+  constraints exist
+- **Change storage class** if PVC provisioning fails:
+  ```bash
+  # Check current storage class status
+  kubectl get storageclass
+  kubectl describe storageclass <current-storage-class>
+  
+  # Look for PVC provisioning errors
+  kubectl describe pvc <pvc-name> -n <namespace>
+  
+  # If seeing "no volume provisioner" or similar errors,
+  # specify a working storage class in VM spec:
+  # spec.dataVolumeTemplates[].spec.pvc.storageClassName:
+  #   <working-class>
+  ```
+
+## Prevention
+
+1. **Resource Planning:**
+   - Monitor cluster resource utilization
+   - Set appropriate resource requests/limits on VMs
```

Comment:
```suggestion
Set appropriate VM guest resources in the VM domain guest spec.
```
I would avoid suggesting working with resource requests or/and limits because those are usually 
auto-calculated by the controller upon VM pod rendering.  The recommended fields are `spec.template.spec.domain.guest.memory` and `spec.template.spec.domain.guest.cpu`